include/openmc/random_ray/flat_source_domain.h

```diff
@@ -0,0 +1,122 @@
+#ifndef OPENMC_RANDOM_RAY_FLAT_SOURCE_DOMAIN_H
+#define OPENMC_RANDOM_RAY_FLAT_SOURCE_DOMAIN_H
+
+#include <vector>
+
+#include "openmc/openmp_interface.h"
+#include "openmc/position.h"
+
+namespace openmc {
+
+/*
+ * The FlatSourceDomain class encompasses data and methods for storing
+ * scalar flux and source region for all flat source regions in a
+ * random ray simulation domain.
+ */
+
+class FlatSourceDomain {
```

Comment:
The `FlatSourceDomain` class (as it is now) controls all flat source regions, so I used the "domain" moniker to try to indicate that it was the collection of all flat source regions. In an upcoming PR, there will be the introduction of a `FlatSourceRegion` class that just represents a single FSR, and the `FlatSourceDomain` will own all of them. Basically, my thinking is that it is good to have a class that is responsible for both storing all the FSRs and providing the functions that iterate/update over all FSRs. If you have ideas for how this all might be structured better, it would likely be best to wait for the fixed source + FSR mesh subdivision PR as there is already major restructuring required for those capabilities.